plugins/splitbutton/plugin.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+( function() {
+	'use strict';
+
+	function getLastActiveCommands( editor, allItems ) {
+		var activeItem = null;
+		for ( var i in allItems ) {
+			if ( editor.getCommand( allItems[ i ].command ).state === CKEDITOR.TRISTATE_ON ) {
+				activeItem = allItems[ i ];
+			}
+		}
+		return activeItem;
+	}
+
+	function createButtons( editor, definition ) {
+		var allItems = definition.items,
+			properties = { items: {}, buttons: {}, groups: {} },
+			i,
+			item,
+			previousButton,
+			defaultButton;
+
+		for ( i in allItems ) {
+			item = allItems[ i ];
+
+			if ( !item.group ) {
+				item.group = definition.label + '_default';
+			}
+
+			if ( properties.groups[ item.group ] === undefined ) {
+				properties.groups[ item.group ] = 0;
+			} else {
+				properties.groups[ item.group ]++;
+			}
+			item.order = properties.groups[ item.group ];
+
+			if ( !item.id ) {
+				item.id = item.command + item.order;
+			}
+
+			item.role = 'menuitemcheckbox';
+
+			item.onClick = function() {
+				editor.execCommand( this.command );
+			};
+
+			editor.getCommand( item.command ).on( 'state', function() {
+				var activeButton = getLastActiveCommands( editor, allItems ),
+					previousId = previousButton && properties.buttons[ previousButton.id ]._.id;
+
+				if ( previousId ) {
+					CKEDITOR.document.getById( previousId ).setStyle( 'display', 'none' );
+					properties.buttons[ previousButton.id ].hidden = true;
+				}
+
+				if ( activeButton ) {
+					CKEDITOR.document.getById( properties.buttons[ defaultButton.id ]._.id ).setStyle( 'display', 'none' );
+					properties.buttons[ defaultButton.id ].hidden = true;
+
+					CKEDITOR.document.getById( properties.buttons[ activeButton.id ]._.id ).removeStyle( 'display' );
+					properties.buttons[ activeButton.id ].hidden = false;
+					previousButton = activeButton;
+				} else {
+					CKEDITOR.document.getById( properties.buttons[ defaultButton.id ]._.id ).removeStyle( 'display' );
```

Comment:
This code is the same as code above and similar to code, in which `display` is set to `none`. It'd move it to separate, dedicated function.